packages/@expo/cli/src/export/exportAssets.ts

```diff
@@ -42,17 +73,35 @@ export async function resolveAssetBundlePatternsAsync<T extends ExpoConfig>(
       const shouldBundle = shouldBundleAsset(asset, fullPatterns);
       if (shouldBundle) {
         debug(`${shouldBundle ? 'Include' : 'Exclude'} asset ${asset.files?.[0]}`);
-        return asset.fileHashes.map(
-          (hash) => 'asset_' + hash + ('type' in asset && asset.type ? '.' + asset.type : '')
-        );
+        return asset.fileHashes.map((hash) => mapAssetHashToAssetString(asset, hash));
       }
       return [];
     })
     .flat();
 
   // The assets returned by the RN packager has duplicates so make sure we
   // only bundle each once.
-  (exp as any).bundledAssets = [...new Set(allBundledAssets)];
+  return new Set(allBundledAssets);
+}
+
+/**
+ * Resolves the assetBundlePatterns from the manifest and returns a list of assets to bundle.
+ *
+ * @modifies {exp}
+ */
+export async function resolveAssetPatternsToBeBundledAsync<T extends ExpoConfig>(
+  projectRoot: string,
+  exp: T,
+  assets: Asset[]
+): Promise<T & { bundledAssets?: Set<string> }> {
```

Comment:
Would probably be good to add `bundledAssets` to the types if we use it or include it in any of the exported JSON. It looks like it is used in legacy manifest accessors in the native layer, but not in modern manifests (yet, or depending on your approach maybe never?).

If this is used in CLI code only, it might be better to just have this method return the list/set rather than mutating the exp object.

Reply:
I will take a look at just returning the set.